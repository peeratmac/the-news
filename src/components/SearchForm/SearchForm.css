input {
  border: none;
  font-size: 30px;
  font-weight: bold;
  margin: 20px;
}

input:focus {
  outline: none;
}
