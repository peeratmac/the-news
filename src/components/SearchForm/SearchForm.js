import React, { Component } from 'react';
import './SearchForm.css';
import PropTypes from 'prop-types';
class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      searchInputQuery: ''
    };
  }

  handleSearchNews = event => {
    this.setState({ searchInputQuery: event.target.value });
    this.props.searchNews(event.target.value);
  };

  render() {
    return (
      <form>
        <input
          onChange={event => this.handleSearchNews(event)}
          name='search-input-query'
          value={this.state.searchInputQuery}
          type='text'
          placeholder='Search The News Here'
        />
      </form>
    );
  }
}

export default SearchForm;

SearchForm.propTypes = {
  searchNews: PropTypes.func.isRequired
};
