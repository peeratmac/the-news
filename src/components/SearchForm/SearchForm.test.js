import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  it('should match the SearchForm Snapshot', () => {
    const mockSearchNews = jest.fn();
    const wrapper = shallow(<SearchForm searchNews={mockSearchNews} />);
    expect(wrapper).toMatchSnapshot();
  });
});
