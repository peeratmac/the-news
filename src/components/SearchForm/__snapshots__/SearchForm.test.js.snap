// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the SearchForm Snapshot 1`] = `
<form>
  <input
    name="search-input-query"
    onChange={[Function]}
    placeholder="Search The News Here"
    type="text"
    value=""
  />
</form>
`;
